unit-tests/unit-tests-hdr.cpp

```diff
@@ -309,276 +309,312 @@ TEST_CASE("HDR Config while Streaming", "[hdr][live][using_pipeline]")
+            {
+                rs2::frameset data = pipe.wait_for_frames();
+                rs2::depth_frame out_depth_frame = data.get_depth_frame();
 
-        // initializing the merging filter
-        rs2::hdr_merge merging_filter;
+                if (out_depth_frame.supports_frame_metadata(RS2_FRAME_METADATA_SEQUENCE_ID))
+                {
+                    auto seq_id = out_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_SEQUENCE_ID);
+                    long long frame_counter = out_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_FRAME_COUNTER);
+                    long long frame_exposure = out_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_EXPOSURE);
+                    long long frame_ts = out_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_FRAME_TIMESTAMP);
+                    //std::cout << "depth - seq_id = " << seq_id << ", frame counter = " << frame_counter << ", exposure = " << frame_exposure << ", ts = " << frame_ts << std::endl;
 
-        int num_of_iterations_in_serie = 10;
-        int first_series_last_merged_ts = -1;
-        for (int i = 0; i < num_of_iterations_in_serie; ++i)
-        {
-            rs2::frameset data = pipe.wait_for_frames();
-            rs2::depth_frame out_depth_frame = data.get_depth_frame();
-
-            auto seq_id = out_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_SEQUENCE_ID);
-            long long frame_counter = out_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_FRAME_COUNTER);
-            long long frame_exposure = out_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_EXPOSURE);
-            long long frame_ts = out_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_FRAME_TIMESTAMP);
-            //std::cout << "depth - seq_id = " << seq_id << ", frame counter = " << frame_counter << ", exposure = " << frame_exposure << ", ts = " << frame_ts << std::endl;
-
-            // merging the frames from the different HDR sequence IDs 
-            auto merged_frameset = merging_filter.process(data);
-            auto merged_depth_frame = merged_frameset.as<rs2::frameset>().get_depth_frame();
-
-            seq_id = merged_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_SEQUENCE_ID);
-            frame_counter = merged_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_FRAME_COUNTER);
-            frame_exposure = merged_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_EXPOSURE);
-            frame_ts = merged_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_FRAME_TIMESTAMP);
-            //std::cout << "merged - seq_id = " << seq_id << ", frame counter = " << frame_counter << ", exposure = " << frame_exposure << ", ts = " << frame_ts << std::endl;
-
-            if (i == (num_of_iterations_in_serie - 1))
-                first_series_last_merged_ts = frame_ts;
-        }
+                    // merging the frames from the different HDR sequence IDs 
+                    auto merged_frameset = merging_filter.process(data);
```

Comment:
Add a REQUIRE check that at least one merged frame exists before going to phase 2

Reply:
Done